### PR TITLE
Read from binary stream

### PR DESCRIPTION
**Pull request by j-brady**
Hi Jonathan,
Thanks for making nmrglue! It's such a wonderful contribution to the NMR community. I have been using it for my work ever since you published your JBNMR paper and it has really empowered my NMR data analyses.

I have a use case whereby I need (want) to read NMRPipe data stored in memory rather than directly from disk. I've added this capability to the `nmrglue.pipe.read` function which can now accept `io.BytesIO` and `bytes` objects containing NMRPipe data. I have written tests which are passing. I think this change should remain entirely compatible with previous versions. Would you be willing to incorporate it? I'm also open to suggestions.

### CODE REVIEW

**Comment by kaustubhmote** [comment]
@j-brady thanks! This works as advertised and will be a nice addition to nmrglue. I will still wait for a comment from @jjhelmus on type annotations before merging this.

**Review by kaustubhmote — COMMENTED** [review]
This works nicely, except I am not sure about the type annotations. Will you be OK removing them for now? We can probably revisit once the syntax is settled.

File: nmrglue/fileio/pipe.py [hunk 1]
```diff
@@ -481,23 +483,27 @@ def dic2datetime(dic):
 ################
 
 
-def read(filename):
+def read(filename: Union[str, bytes, io.BytesIO]) -> Tuple[dict, np.array]:
```
[the remark this answers is not in the document]
Comment (j-brady) [reply nmrglue/fileio/pipe.py:486]: Hi @kaustubhmote , thanks for reviewing my code! I'll remove the type annotations (probably got a little carried away!). Are you planning on implementing support for Path objects? I wouldn't mind giving that a go as I often have to deal with converting Path objects to strings when using the `nmrglue.fileio` functions which is not ideal.

[the remark this answers is not in the document]
Comment (j-brady) [reply nmrglue/fileio/pipe.py]: Type annotations (and unused imports) are now removed.

**Pull request merged**